docs/source/class_basics.rst

```diff
@@ -151,7 +151,101 @@ concrete. As with normal overrides, a dynamically typed method can
 implement a statically typed abstract method defined in an abstract
 base class.
 
+.. _protocol-types:
+
+Protocols and structural subtyping
+**********************************
+
+Mypy provides support for structural subtyping and protocol classes.
```

Comment:
It would be better to start with some motivation and a discussion of the difference between inheritance-based (nominal) subtyping and structural subtyping, and then follow on with the example. Maybe also mention how this is related duck typing, which is a term many Python programmers are familiar with. The term structural subtyping is probably not as widely known.

Another idea would pick an initial example that just uses a pre-existing protocol such as `Sized` and defines  a sized user-defined class that doesn't explicitly subclass `Sized`. Once we've explained this, we could have another example that defines a custom protocol. I suspect that using a predefined protocol is more common than defining a custom protocol, and it's a simpler use case so it may make sense to start with that.